docker/docker.go

```diff
@@ -444,7 +450,7 @@ func parseImageResponse(resp *http.Response, image *Image) error {
 			return fmt.Errorf("docker Registry responded with unsupported Content-Type (%v). dump error=%+v", contentType, dumpErr)
 		}
 
-		return fmt.Errorf("docker Registry responded with unsupported Content-Type: response=%s", string(dump))
+		return fmt.Errorf("docker Registry responded with unsupported Content-Type: response=%s. %w", string(dump), types.ErrorUnauthorized)
```

Comment:
@Erezf-p why it's an ErrorUnauthorized? Youv'e check for http.StatusUnauthorized in the beginning right?

Reply:
This is what I thought as well. But when I tested it (image with no pull secret) I got the error from here. So I added it also here

Reply:
What was the status code in this case? Why not adding it to the switch case and return a specific error? 
I don't think the default error should be unauthorized, it should be unknown with the error status printed that we will know how to update the switch to give more relevant error.